tests/test_ir.py

```diff
@@ -518,6 +520,64 @@ def test_iteration_parallelism(self, exprs, atomic, parallel):
         assert all(i.is_Parallel for i in iters if i.dim.name in parallel)
         assert all(not i.is_Parallel for i in iters if i.dim.name not in parallel)
 
+    @pytest.mark.parametrize('exprs,atomic,parallel', [
+        (['Inc(u[gp[p, 0]+rx, gp[p, 1]+ry, gp[p, 2]+rz],'
+          ' u[gp[p, 0]+rx, gp[p, 1]+ry, gp[p, 2]+rz] + cx*cy*cz*src)'],
+         ['p', 'rx', 'ry', 'rz'], []),
+        (['Eq(rcv, 0)', 'Inc(rcv, rcv + cx*cy*cz)'],
+         ['rx', 'ry', 'rz'], ['time', 'p']),
+        (['Eq(v.forward, u+1)', 'Eq(rcv, 0)',
+          'Inc(rcv, rcv + v[t, gp[p, 0]+rx, gp[p, 1]+ry, gp[p, 2]+rz]*cx*cy*cz)'],
+         ['rx', 'ry', 'rz'], ['x', 'y', 'z', 'p']),
+        (['Eq(v.forward, v[t+1, x+1, y, z]+v[t, x, y, z]+v[t, x+1, y, z])'],
+         [], ['y', 'z']),
+        (['Eq(v.forward, v[t+1, x-1, y, z]+v[t, x, y, z]+v[t, x-1, y, z])'],
+         [], ['y', 'z']),
+        (['Eq(v.forward, v[t+1, x, y+1, z]+v[t, x, y, z]+v[t, x, y+1, z])'],
+         [], ['x', 'z']),
+        (['Eq(v.forward, v[t+1, x, y-1, z]+v[t, x, y, z]+v[t, x, y-1, z])'],
+         [], ['x', 'z']),
+        (['Eq(v.forward, v[t+1, x, y, z+1]+v[t, x, y, z]+v[t, x, y, z+1])'],
+         [], ['x', 'y']),
+        (['Eq(v.forward, v[t+1, x, y, z-1]+v[t, x, y, z]+v[t, x, y, z-1])'],
+         [], ['x', 'y'])
+    ])
+    def test_iteration_parallelism_3d(self, exprs, atomic, parallel):
```

Comment:
Do we really need to distinguish between 2d and 3d tests?

Reply:
Well, it was only to save the one that was already there from having to declare a second "grid" and an extra set of `Function` objects for the 3D cases.  Will happily put it all into a single function if you prefer.

Reply:
no, OK, that I got. What I meant was, what are we testing with 3d tests that we're not catching with 2d ones? just a curiosity